include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -130,6 +130,22 @@ __find_start_point(const _Rng1& __rng1, const _Index __rng1_from, _Index __rng1_
     return _split_point_t<_Index>{*__res, __index_sum - *__res + 1};
 }
 
+template <typename _Rng1, typename _Rng2, typename _value_t_rng1 = oneapi::dpl::__internal::__value_t<_Rng1>,
+          typename _value_t_rng2 = oneapi::dpl::__internal::__value_t<_Rng2>>
+constexpr auto
+__can_use_ternary_op(int) -> decltype(true ? std::declval<_value_t_rng1>() : std::declval<_value_t_rng2>(),
+                                      std::true_type{})
+{
+    return {};
+}
+
+template <typename _T1, typename _T2>
+constexpr auto
+__can_use_ternary_op(...) -> std::false_type
```

Comment:
Whether should be long instead of `...` because "error : SYCL kernel cannot call a variadic function"?

Reply:
It was on caller code side because I wrote `__can_use_ternary_op<_Rng1, _Rng2>()` without any arguments

Reply:
So what you propose for now&
To rewrite as
```C++
template <typename _Rng1, typename _Rng2>
constexpr static bool __can_use_ternary_op_v = decltype(__can_use_ternary_op<_Rng1, _Rng2>(int{0}))::value;
```
?

Reply:
Yes I think so, and also use `long` instead of `...`.

Reply:
Done.